Helm: Update sizing plans

dimitarvdimitrov opened the pull request:
<!--  Thanks for sending a pull request!  Before submitting:

1. Read our CONTRIBUTING.md guide
2. Rebase your PR if it gets out of sync with main
-->

#### What this PR does


Updates the sizing plans for Helm and add small.yaml and large.yaml to the rendered manifests.

#### Which issue(s) this PR fixes or relates to

Fixes #2890 

#### Checklist

- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

pracucci commented:
The CHANGELOG has just been cut to prepare for the next Mimir release. Please rebase `main` and eventually move the CHANGELOG entry added / updated in this PR to the top of the CHANGELOG document. Thanks!

krajorama commented on operations/helm/charts/mimir-distributed/values.yaml:
```diff
@@ -530,9 +530,6 @@ distributor:
 
```
Ruler, alertmanager stateless deployment mode also has:
```
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxSurge: 0
      maxUnavailable: 1
```
Let's remove that as well.

Also query-scheduler, but we only support two at the moment, so that's probably fine.

krajorama replied:
Thank you for the explanation.

krajorama commented on operations/helm/charts/mimir-distributed/small.yaml:
```diff
@@ -1,62 +1,68 @@
   persistentVolume:
     enabled: true
-  replicas: 3
+  replicas: 1
```
I think we should have 2. I'm worried that if the single alertmanager is shut down, but k8s cannot schedule the new one, the user is stuck without an alertmanager.

krajorama commented on operations/helm/charts/mimir-distributed/capped-small.yaml:
```diff
@@ -1,65 +1,72 @@
   persistentVolume:
     enabled: true
-  replicas: 3
+  replicas: 1
```
I think we should have 2. I'm worried that if the single alertmanager is shut down, but k8s cannot schedule the new one, the user is stuck without an alertmanager.

krajorama commented on operations/helm/charts/mimir-distributed/CHANGELOG.md:
```diff
@@ -29,12 +29,14 @@ Entries should include a reference to the Pull Request that introduced the chang
 * [CHANGE] Nginx: replace topology key previously used in `podAntiAffinity` (`failure-domain.beta.kubernetes.io/zone`) with a different one `topologySpreadConstraints` (`kubernetes.io/hostname`). #2722
 * [CHANGE] Use `topologySpreadConstraints` instead of `podAntiAffinity` by default. #2722
 * [CHANGE] Ingresses for the GEM gateway and nginx will no longer render on Kubernetes versions <1.19. #2872
+* [CHANGE] **breaking change** Reduce the number of ingesters in small.yaml and capped-small.yaml form 4 to 3. This should be more accurate size for the scale of 1M AS. Before upgrading refer to [Scaling down ingesters](https://grafana.com/docs/mimir/latest/operators-guide/run-production-environment/scaling-out/#scaling-down-ingesters) to scale down `ingester-3`. Alternatively override the number ...
```
```suggestion
* [CHANGE] **breaking change** Reduce the number of ingesters in small.yaml and capped-small.yaml from 4 to 3. This should be a more accurate size for the scale of 1M AS. Before upgrading refer to [Scaling down ingesters](https://grafana.com/docs/mimir/latest/operators-guide/run-production-environment/scaling-out/#scaling-down-ingesters) to scale down `ingester-3`. Alternatively override the number of ingesters to 4. #3035
```

dimitarvdimitrov commented:
thanks for the review. I think I addressed all your comments. 

I was intending to first merge https://github.com/grafana/mimir/pull/3029 so the diff here is easier to review. Do you mind reviewing that too?

krajorama commented on operations/helm/charts/mimir-distributed/CHANGELOG.md:
````diff
@@ -76,6 +78,7 @@ Entries should include a reference to the Pull Request that introduced the chang
                 topologyKey: 'kubernetes.io/hostname'
      ```
 * [CHANGE] Ingresses for the GEM gateway and nginx will no longer render on Kubernetes versions <1.19. #2872
+* [CHANGE] **breaking change** Reduce the number of ingesters in small.yaml and capped-small.yaml from 4 to 3. This should be a more accurate size for the scale of 1M AS. Before upgrading refer to [Scaling down ingesters](https://grafana.com/docs/mimir/latest/operators-guide/run-production-environment/scaling-out/#scaling-down-ingesters) to scale down `ingester-3`. Alternatively override the numb ...
````
```suggestion
```

krajorama commented on operations/helm/charts/mimir-distributed/CHANGELOG.md:
```diff
@@ -25,10 +25,12 @@ Entries should include a reference to the Pull Request that introduced the chang
   Mimir under the `metamonitoring` tenant; this enhancement does not apply to GEM. #3176
 * [ENHANCEMENT] Improve default rollout strategies. Now distributor, overrides_exporter, querier, query_frontend, admin_api, gateway, and graphite components can be upgraded more quickly and also can be rolled out with a single replica without downtime. #3029
 * [ENHANCEMENT] Metamonitoring: make scrape interval configurable. #2945
+* [ENHANCEMENT] Update sizing plans (small.yaml, large.yaml, capped-small.yaml, capped-large.yaml). These reflect better how we recommend running Mimir and GEM in production. #3035
```
I think this should be  a `[CHANGE]` and add a disclaimer that some replica counts and resource requests have been reduced!

dimitarvdimitrov replied:
they have all changed. Should I list all the changes? I'd rather just link to the diff from this PR. Do you think that's ok @krajorama ?

krajorama replied:
Yeah, I think linking to the PR is ok, also ppl can use helm diff...

krajorama replied:
Mention that some replica counts and resource requests and limits are reduced!

dimitarvdimitrov replied:
done in https://github.com/grafana/mimir/pull/3035/commits/c064706717d32c7465b1fc6093242e3fff6dfebf

krajorama commented on operations/helm/charts/mimir-distributed/CHANGELOG.md:
````diff
@@ -96,6 +98,7 @@ Entries should include a reference to the Pull Request that introduced the chang
                 topologyKey: 'kubernetes.io/hostname'
      ```
 * [CHANGE] Ingresses for the GEM gateway and nginx will no longer render on Kubernetes versions <1.19. #2872
+* [CHANGE] **breaking change** Reduce the number of ingesters in small.yaml and capped-small.yaml from 4 to 3. This should be a more accurate size for the scale of 1M AS. Before upgrading refer to [Scaling down ingesters](https://grafana.com/docs/mimir/latest/operators-guide/run-production-environment/scaling-out/#scaling-down-ingesters) to scale down `ingester-3`. Alternatively override the numb ...
````
```suggestion
```
